### PR TITLE
docs: improve module docs

### PR DESCRIPTION
**Pull request by cosmic-vagabond**
### What has Changed?

Improve documentations for:
* AMM
* Clock
* Epochs
* Parameter
* StableStake

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/elys-network/elys/blob/main/docs/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#testing)
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### Deployment Notes

Are there any specific considerations to take into account when deploying these changes? This may include new dependencies, scripts that need to be executed, or any aspects that can only be evaluated in a deployed environment.

### Screenshots and Videos

Please provide any relevant before and after screenshots by uploading them here. Additionally, demo videos can be highly beneficial in demonstrating the process.

### CODE REVIEW

File: x/amm/spec/02_mechanism.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+The Elys Network AMM module implements an Automated Market Maker (AMM) using various types of liquidity pools. It supports the following types of pools:
+
+1. **AMM Pools**: Non-oracle pools with liquidity centered around a specific spot price, designed for assets with significant price variation.
+2. **Oracle Pools**: Oracle pools with liquidity centered around an oracle price, designed for assets with stable prices.
+
+This section explains the fundamental mechanism of the AMM module and provides an overview of how the module's code is structured to support both types of pools.
+
+## Pool
+
+### Creation of Pool
```
Comment (amityadav0) [inline x/amm/spec/02_mechanism.md:12]: is it possible to add more detail about oracle and non-oracle pool equations ?

Comment (amityadav0) [reply x/amm/spec/02_mechanism.md:12]: what if in oracle pool, liquidity tokens lose balance w.r.t oracle prices ?

File: x/stablestake/spec/02_mechanism.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
+<!--
+order: 2
+-->
+
+# Mechanism
+
+## Overview
+
+The StableStake module facilitates leveraged liquidity provision (LP) positions by allowing users to borrow funds against their collateral. This process involves interactions between the user, a stable stake pool, and an Automated Market Maker (AMM) pool. The module ensures efficient and secure fund management, enabling users to maximize their liquidity provisioning potential.
+
```
Comment (amityadav0) [inline x/stablestake/spec/02_mechanism.md:10]: User can also bond and unbond their USDC to get interest rates, is that right ?

File: x/amm/spec/02_mechanism.md (hunk 1 above)
Comment (cosmic-vagabond) [reply x/amm/spec/02_mechanism.md:18]: @amityadav0 I added some additional content to cover the oracle pool mechanism: https://github.com/elys-network/elys/pull/640/commits/0efeecc1ebcbe0b79da0d41f731820ad742e3c1c

**Review by amityadav0 — APPROVED** [review]
LGTM

**Pull request merged**